### PR TITLE
Expand on the definition of Flags

### PR DESCRIPTION
**Pull request by youngmit**
This is intended to address #163

### CODE REVIEW

File: armi/reactor/flags.py [hunk 1]
```diff
@@ -13,57 +13,82 @@
...
-ignored to avoid defining hundreds of flags for assemblies with many pins in them (e.g.
-in pin-level depletion cases). Use ``FUEL A`` or ``FUEL B`` to distinguish if necessary.
-
-Code modules that check the names of objects must use valid flags only.
-
-Things that flags are used for include:
+Handles *flags* that formally bind certain categories to reactor parts.
+
+``Flags`` are used to formally categorize the various ``ArmiObject`` objects that make
+up a reactor model. These categories allow parts of the ARMI system treat different
```
Comment (jakehader) [inline armi/reactor/flags.py:19]: `to treat`

File: armi/reactor/flags.py [hunk 2]
```diff
@@ -13,57 +13,82 @@
...
+
+    Migrating the `meaning` of Flags into settings will allow us to better document how
+    those Flags/settings affect ARMI's behavior.
+
+    As mentioned above, plenty of code still hard-codes Flag ``TypeSpecs``, and certain
+    Flags do not follow the `what something is` convention. Future work should improve
+    upon this as possible.
+
+
+Things that Flags are used for include:
```
Comment (jakehader) [inline armi/reactor/flags.py:69]: Should we consider adding some discussion around the use of settings to define the behavior? Also, this is good documentation, but it seems like it should be discoverable in the docs as well. Should we link to this discussion?

Comment (youngmit) [reply armi/reactor/flags.py:69]: Yeah there should probably be something in the main doc/ tree pointing to this...

**Review by jakehader — CHANGES_REQUESTED** [review]
Add links to the flags description in the main docs.

**Review by jakehader — APPROVED** [review]
🔥

**Comment by youngmit** [comment]
Fixes #163

**Pull request merged**